bin/prx-dovetail-regions.rb

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env ruby
+# View/update which Dovetail Router regions DNS is pointing at.
+
+require "bundler/inline"
+
+gemfile do
+  source "https://rubygems.org"
+  gem "awesome_print"
+  gem "aws-sdk-route53"
+  gem "prx-ruby-aws-creds"
+  gem "slop"
+end
+
+OPTS = Slop.parse do |o|
+  o.string "-p", "--profile", "AWS profile", default: "prx-legacy-route53"
+  o.string "-e", "--environment", 'E.g. "production" or "staging"', required: true
+  o.string "-c", "--change", 'Change dovetail regions (e.g., "us-east-1" or "all")', default: nil
+  o.on "-h", "--help" do
+    puts o
+    exit
+  end
+end
+
+ZONE_NAME = "prx.tech"
+ALL_REGIONS = %w[us-east-1 us-west-2]
+if %w[stag staging].include?(OPTS[:environment].to_s.downcase)
+  ALIAS = "dovetail-router.staging.u.#{ZONE_NAME}"
+  GROUP = "dovetail-router-alb-latency-group.stag.#{ZONE_NAME}"
+  REGIONS = ALL_REGIONS.map { |r| [r, "dovetail-router.staging.#{r}.#{ZONE_NAME}"] }.to_h
+elsif %w[prod production].include?(OPTS[:environment].to_s.downcase)
+  ALIAS = "dovetail-router.u.#{ZONE_NAME}"
+  GROUP = "dovetail-router-alb-latency-group.prod.#{ZONE_NAME}"
+  REGIONS = ALL_REGIONS.map { |r| [r, "dovetail-router.#{r}.#{ZONE_NAME}"] }.to_h
+else
+  abort "Invalid environment".red
+end
+
+CHANGE =
+  if OPTS[:change] == "all"
+    GROUP
+  elsif REGIONS.key?(OPTS[:change])
+    REGIONS[OPTS[:change]]
+  elsif OPTS[:change]
+    abort "Invalid change region".red
+  end
+
+# check if a route53 aws profile exists
+creds =
+  begin
+    PrxRubyAwsCreds.client_credentials
+  rescue => e
+    puts "Unable to get AWS client credentials!".red
+    puts "\nDid you add a #{"[profile prx-legacy-route53]".yellow} to your ~/.aws/config?"
+    exit 1
+  end
+
+# find hosted zone
+client = Aws::Route53::Client.new(credentials: creds, retry_mode: "adaptive")
```

Comment:
Despite having `region = us-east-1` set in my `prx-legacy-route53` profile, I had to add `region: "us-east-1"` to this, otherwise I was getting a NO REGION error

Reply:
Oh, hrm... I assumed route53 had no region?  I guess for an SDK it does.  Or at least to authenticate.